sdcclient/_secure.py

```diff
@@ -612,79 +628,81 @@ def list_policies(self):
-
-        try:
-            json.loads(priorities_json)
-        except Exception as e:
-            return [False, "priorities json is not valid json: {}".format(str(e))]
-
-        res = requests.put(self.url + '/api/policies/priorities', headers=self.hdrs, data=priorities_json, verify=self.ssl_verify)
+        res = requests.get(self.url + '/api/v2/policies/{}'.format(id), headers=self.hdrs, verify=self.ssl_verify)
         return self._request_result(res)
 
-    def get_policy(self, name):
+    def add_policy(self, name, description, rule_names=[], actions=[], scope=None, severity=0, enabled=True, notification_channels=[]):
         '''**Description**
-            Find the policy with name <name> and return its json description.
+            Add a new policy.
 
         **Arguments**
-            - name: the name of the policy to fetch
+            - name: A short name for the policy
+            - description: Description of policy
+            - rule_names: Array of rule names. (They must be names instead of ids, as the rules list view is by name, to account for multiple rules having the same name).
+            - actions: It can be a stop, pause and/or capture action
+            - scope: Where the policy is being applied- Container, Host etc.. (example: "container.image.repository = sysdig/agent")
+            - enabled: True if the policy should be considered
+            - severity: How severe is this policy when violated. Range from 0 to 7 included.
+            - notification_channels: ids of the notification channels to subscribe to the policy
 
         **Success Return Value**
-            A JSON object containing the description of the policy. If there is no policy with
-            the given name, returns False.
-
-        **Example**
-            `examples/get_policy.py <https://github.com/draios/python-sdc-client/blob/master/examples/get_policy.py>`_
-
+            The string "OK"
         '''
-        res = requests.get(self.url + '/api/policies', headers=self.hdrs, verify=self.ssl_verify)
-        if not self._checkResponse(res):
-            return [False, self.lasterr]
-
-        policies = res.json()["policies"]
-
-        # Find the policy with the given name and return it.
-        for policy in policies:
-            if policy["name"] == name:
-                return [True, policy]
-
-        return [False, "No policy with name {}".format(name)]
+        policy = {
+            "name": name,
+            "description": description,
+            "ruleNames": rule_names,
+            "actions": actions,
+            "scope": scope,
+            "severity": severity,
+            "enabled": enabled,
+            "notificationChannelIds": notification_channels
+        }
+        res = requests.post(self.url + '/api/v2/policies', headers=self.hdrs, data=json.dumps(policy), verify=self.ssl_verify)
+        return self._request_result(res)
 
-    def add_policy(self, policy_json):
+    def add_policy_json(self, policy_json):
```

Comment:
What was the rationale behind adding the `_json` postfix? I'm asking because almost all functions in the libraries deal with JSON one way or another (parameters and outputs), so I wonder if the postfix is unnecessary.

Reply:
There are now two ways of adding a policy, one is passing the params separatelly (see [add_policy](https://github.com/draios/python-sdc-client/pull/99/files#diff-e6dd2abfa2a06413286a7994889acedcR675)) another is passing a json. This is why I added the `_json`, I need a way to differentiate the two methods. I'm happy to hear ideas on how to improve the naming.